modules/account-lib/src/coin/hbar/transaction.ts

```diff
@@ -189,23 +226,42 @@ export class Transaction extends BaseTransaction {
    */
   loadInputsAndOutputs(): void {
     const txJson = this.toJson();
-    if (txJson.to && txJson.amount) {
-      this._outputs = [
-        {
-          address: txJson.to,
-          value: txJson.amount,
-          coin: txJson.tokenName || this._coinConfig.name,
-        },
-      ];
+    const instruction = txJson.instructionsData;
+    const outputs: Entry[] = [];
+    const inputs: Entry[] = [];
 
-      this._inputs = [
-        {
+    switch (instruction?.type) {
+      case HederaTransactionTypes.Transfer:
+        let totalAmount = new BigNumber(0);
+        instruction.params.recipients.map((recipient) => {
+          totalAmount = totalAmount.plus(recipient.amount);
+          outputs.push({
+            address: recipient.address,
+            value: recipient.amount,
+            coin: recipient.tokenName || this._coinConfig.name,
+          });
+        });
+        inputs.push({
           address: txJson.from,
-          value: txJson.amount,
-          coin: txJson.tokenName || this._coinConfig.name,
-        },
-      ];
+          value: totalAmount.toString(),
+          coin: instruction.params.tokenName || this._coinConfig.name,
+        });
+        break;
+
+      case HederaTransactionTypes.TokenAssociateToAccount:
+        instruction.params.tokens.map((tokenName) => {
+          const tokenEntry: Entry = {
+            address: instruction.params.accountId,
+            value: TRANSFER_AMOUNT_UNKNOWN_TEXT,
```

Comment:
shouldn't we be able to get this from the `txJson`?

Reply:
It's from txJson but in txJson `instructionData`, it has different format from tokenEntry, so this is just adjustment for the tokenEntry interface

Reply:
i meant just the `value` amount

Reply:
currently `instructionData` for `token associate` has `accountId` for owner and `tokens` which is list of `tokenName`s
I think adding an unknown value of amount in the `tokens` will be weird?

Reply:
ohhhh this is for token association. lets use 0 instead?